### ISSUE TITLE
 [Paladin] Blood Corruption/Holy Vengeance

### ISSUE THREAD

**Issue by Bordiran** [SunwellTracker/issues#1496]
Decription: 
Blood Corruption (Horda) / Holy Vengeance (Ally) should convert your dmg as refreshing, after the percentage of dmg trinkets should increase, how the trinket ends should fall, the same applies to the% increase in dmg from the wings, etc.

How it works: 
Blood Corruption / Holy Vengeance gets statistics / boost dmg at the time of imposing the fifth stack and holds them until the stacks fall.

How it should work: 
Blood Corruption / Holy Vengeance should download statistics / boost dmg every refreshing

Source (you should point out proofs of your report, please give us some source): 
Logi Sunwellowe: http://worldoflogs.com/reports/bqb7hdd1gkc59lqz/log/?s=792&e=953
Query: [{"spellNames": ["Blood Corruption"], "actorNames": ["Elaatu"]}, {"spellNames": ["Avenging Wrath"], "actorNames": ["Elaatu"]}, {"spellNames": ["Paragon"], "actorNames": ["Elaatu"]}]

[20:23:35.482] Elaatu casts Avenging Wrath
[20:23:35.483] Elaatu gains Avenging Wrath from Elaatu
[20:23:35.741] Lord Jaraxxus afflicted by Blood Corruption from Elaatu
[20:23:37.889] Lord Jaraxxus afflicted by Blood Corruption (2) from Elaatu
[20:23:38.691] Elaatu Blood Corruption Lord Jaraxxus 848 (R: 44)
[20:23:40.019] Lord Jaraxxus afflicted by Blood Corruption (3) from Elaatu
[20:23:40.019] Elaatu gains Paragon from Elaatu
[20:23:41.717] Elaatu Blood Corruption Lord Jaraxxus 1565 (R: 82)
[20:23:41.795] Elaatu gains Paragon from Elaatu
[20:23:42.128] Lord Jaraxxus afflicted by Blood Corruption (4) from Elaatu
[20:23:44.350] Lord Jaraxxus afflicted by Blood Corruption (5) from Elaatu
[20:23:44.649] Elaatu Blood Corruption Lord Jaraxxus 3277
[20:23:47.664] Elaatu Blood Corruption Lord Jaraxxus 3277
[20:23:50.676] Elaatu Blood Corruption Lord Jaraxxus 3114 (R: 163)
[20:23:53.717] Elaatu Blood Corruption Lord Jaraxxus 3277
[20:23:54.989] Elaatu's Paragon fades from Elaatu
[20:23:55.446] Elaatu's Avenging Wrath fades from Elaatu
[20:23:56.676] Elaatu Blood Corruption Lord Jaraxxus 2950 (R: 327)
[20:23:56.760] Elaatu's Paragon fades from Elaatu
[20:23:59.664] Elaatu Blood Corruption Lord Jaraxxus 3277
[20:24:02.662] Elaatu Blood Corruption Lord Jaraxxus 3277
[20:24:05.702] Elaatu Blood Corruption Lord Jaraxxus 2950 (R: 327)
[20:24:08.718] Elaatu Blood Corruption Lord Jaraxxus 3277
[20:24:11.592] Elaatu Blood Corruption Lord Jaraxxus 3277
[20:24:14.718] Elaatu Blood Corruption Lord Jaraxxus 2950 (R: 327)

Logi Globalowe: http://www.worldoflogs.com/reports/mfxew5mdm3orj505/log/?s=2395&e=2572
Query: [{"spellNames": ["Holy Vengeance"], "actorNames": ["Elovie"]}, {"spellNames": ["Icy Rage"], "actorNames": ["Elovie"]}, {"spellNames": ["Avenging Wrath"], "actorNames": ["Elovie"]}]

[17:37:55.668] Deathbringer Saurfang afflicted by Holy Vengeance from Elovie
[17:37:56.023] Elovie gains Icy Rage from Elovie
[17:37:58.080] Deathbringer Saurfang afflicted by Holy Vengeance (2) from Elovie
[17:37:58.618] Elovie Holy Vengeance Deathbringer Saurfang 1162 (R: 114)
[17:38:00.108] Deathbringer Saurfang afflicted by Holy Vengeance (3) from Elovie
[17:38:01.653] Elovie Holy Vengeance Deathbringer Saurfang 2140
[17:38:03.353] Deathbringer Saurfang afflicted by Holy Vengeance (4) from Elovie
[17:38:04.635] Elovie Holy Vengeance Deathbringer Saurfang 2853
[17:38:04.929] Deathbringer Saurfang afflicted by Holy Vengeance (5) from Elovie
[17:38:06.322] Elovie gains Avenging Wrath from Elovie
[17:38:06.322] Elovie casts Avenging Wrath
[17:38:07.705] Elovie Holy Vengeance Deathbringer Saurfang 4204
[17:38:10.628] Elovie Holy Vengeance Deathbringer Saurfang 3544 (R: 349)
[17:38:11.050] Elovie's Icy Rage fades from Elovie
[17:38:13.647] Elovie Holy Vengeance Deathbringer Saurfang 3245 (R: 319)
[17:38:16.677] Elovie Holy Vengeance Deathbringer Saurfang 3245 (R: 319)
[17:38:19.636] Elovie Holy Vengeance Deathbringer Saurfang 3283 (R: 323)
[17:38:22.609] Elovie Holy Vengeance Deathbringer Saurfang 3692
[17:38:25.635] Elovie Holy Vengeance Deathbringer Saurfang 3544
[17:38:26.336] Elovie's Avenging Wrath fades from Elovie
[17:38:28.673] Elovie Holy Vengeance Deathbringer Saurfang 3077
[17:38:31.657] Elovie Holy Vengeance Deathbringer Saurfang 2768 (R: 272)
[17:38:34.669] Elovie Holy Vengeance Deathbringer Saurfang 3076
[17:38:37.611] Elovie Holy Vengeance Deathbringer Saurfang 2460 (R: 544)
[17:38:40.712] Elovie Holy Vengeance Deathbringer Saurfang 3075

Copied this Bug Report from a Feronis Bug Report, the same bug is present on Angrathar server.

**Comment by Bordiran** [comment]
After the 3% nerf, Retribution is literally useless because they're so bugged and do so little DPS. This needs an immediate fix please :(

**Comment by Bordiran** [comment]
Someone from my guild who played Ret Paladin on Feronis and plays it here says could also be issue with the coefficients on Seal of Vengeance/Corruption.

**Comment by ujellyx** [comment]
Do you have any sources for it working the way it should other than a similar report from another private server?

**Comment by Bordiran** [comment]
This is report from Feronis, it's Sunwell server. Elaatu is the one who made it.

**Comment by ujellyx** [comment]
And how did they prove that it's really a bug on Feronis? Any links that prove, how it really was on retail?

**Comment by Bordiran** [comment]
http://wowwiki.wikia.com/wiki/Blood_Corruption and http://wowwiki.wikia.com/wiki/Holy_Vengeance

**Comment by ujellyx** [comment]
These links don't prove anything. As the damage buffs wear off, the damage of the stacks themselves gets lower/normal again, as it should be. I still don't see a "bug" here, as it behaves exactly like it should.

**Comment by Bordiran** [comment]
Elaatu should have it, he was the one who originally reported it and it got fixed on Feronis. He's usually around the bug tracker.

**Comment by Bordiran** [comment]
https://gyazo.com/8032993368c3e9ea4eadaac5098f1afc This is the change that messed it up and made it really low dmg. Something to do with the Coefficients of Seal of Corruption/Vengeance the proc and DoT.

**Comment by Bordiran** [comment]
As you can see in the logs comparing Sunwell and Retail, the DoT isn't getting dmg increase at all from Avenging Wrath + Trinkets and other damage modifiers. Damage DOT is doing at 5 stacks with Wings + Trinkets is same damage as without in the Sunwell logs. Whereas in the Retail Logs you can see the damage lower when wings + trinkets fade.

**Comment by Bordiran** [comment]
11:52:09.194] Danbtw gains Greatness from Danbtw
[11:52:12.222] Danbtw Blood Corruption VX-001 632 (R: 70)
[11:52:13.209] Danbtw casts Avenging Wrath
[11:52:13.210] Danbtw gains Avenging Wrath from Danbtw
[11:52:15.103] Danbtw Blood Corruption VX-001 1305
[11:52:18.063] Danbtw Blood Corruption VX-001 1631 (R: 181)
[11:52:21.149] Danbtw Blood Corruption VX-001 2587
[11:52:24.210] Danbtw Blood Corruption VX-001 2587
[11:52:24.211] Danbtw's Greatness fades from Danbtw
[11:52:27.189] Danbtw Blood Corruption VX-001 2587
[11:52:30.079] Danbtw Blood Corruption VX-001 2587
[11:52:33.185] Danbtw Blood Corruption VX-001 2587
[11:52:33.207] Danbtw's Avenging Wrath fades from Danbtw
[11:52:36.085] Danbtw Blood Corruption VX-001 2587
[11:52:39.116] Danbtw Blood Corruption VX-001 2587
[11:52:42.112] Danbtw Blood Corruption VX-001 2587

This is from my personal logs on Mimiron hard mode. As you can see, it goes up to 2587 damage per DoT tick at 5 stacks but it's gaining no benefit from my Avenging Wrath or my Trinket procs because when you see Avenging Wrath and Trinkets fade, the damage per tick doesn't change.

**Comment by Bordiran** [comment]
9% Damage increase from Vengeance Talent isn't affecting my Blood Corruption either, I just tested on a Dummy in Orgrimmar. I assume the Crusade Talent and everything that is supposed to modify the damage of Blood Corruption isn't working, which is gimping Ret Paladin DPS so hard.

**Comment by Bordiran** [comment]
Please let me know if any more info or whatever is needed, really need this to be fixed as soon as possible for Retribution Paladins damage for Ulduar Progression.

**Comment by Bordiran** [comment]
Demonic Pact from Demo Lock isn't affecting it either when it should, adding to the list of things not affecting it which is probably everything.

**Comment by Azathothh** [comment]
https://github.com/SunwellTracker/issues/issues/10
It's already reported also your comments are bit misleading:
" it's gaining no benefit from my Avenging Wrath or my Trinket procs" - that's not true
"because when you see Avenging Wrath and Trinkets fade, the damage per tick doesn't change" - it doesn't change because it doesn't update with your current stats - and that's the bug

**Comment by Bordiran** [comment]
I just don't see how my comments can be "misleading" when that's exactly how it is :D

**Comment by Azathothh** [comment]
I don't think my previous post is really that hard to understand...
You were stating that Avenging Wrath or your trinkets DOESN'T work at all on your Blood Corruption (by pasting combatlog where you're stacking it up with greatness and AW active), and like i've already said it's not true. 
Just stack blood corruption to 5 without Avenging Wrath and trinkets -> save results -> stack blood corruption to 5 again but with trinkets and Avanging Wrath used while you're stacking it up -> compare results -> yes, ticks from second attempt are higher because Avenging Wrath and trinkets DOES WORK on Blood Corruption.  (Yes, it also means that if you use Avenging Wrath at beginning of the fight, while stacking up Blood Corruption, this 20% dmg bonus will be applied on Blood Corruption as long as it's up on boss (so most fights in ulduar))
They are misleading because you're implying that theres a bug with Avenging Wrath / trinkets / other stuff not working towards damage of Blood Corruption while it's all working. 

The bug is that Blood Corruption/Holy Vengeance does use stats from the moment when it was applied for entire duration (just like warlock's corruption) instead of having DoT damage updated with current stats after each tick. Things that you're listing in multiple comments are just RESULTS of this bug.
I believe it's much better to fix SOURCE of problem, isn't it?

**Comment by Bordiran** [comment]
That's probably true, I haven't tested that yet. My misunderstanding comes from you saying I was wrong but not explaining why I was wrong, so you explaining why clears that up. Either way it's still a pretty big bug on most fights in Ulduar, especially when you have to restack Blood Corruption with no Avenging Wrath/Trinkets/Demonic Pact, etc. etc. I assume you're right but I'll check on the dummy after raid to confirm.

**Comment by Bordiran** [comment]
Yeah, can confirm. It's snapshotting like Warlock Corruption instead of updating with current stats.

**Comment by Bordiran** [comment]
However, it's interacting strangely with things. I had a Rogue tricks me before I applied my 5th stack of Blood Corruption, then I had a Rogue tricks me before I applied by 1st stack of Blood Corruption then stacked to to 5 before the Tricks ran out and Blood Corruption didn't get the 15% damage increase. It was ticking for 1470 without Tricks but with Avenging Wrath and Trinkets, then when I did the same thing with Tricks (1470x0.15= 220, 1470 + 220 = 1690 dmg per tick) it was still 1470. So Blood Corruption is really awkward at the moment because the whole "snapshotting" thing it does at the moment doesn't even work 100% properly.
